vcr/api/v2/api_test.go

```diff
@@ -540,6 +541,63 @@ func TestWrapper_CreateVP(t *testing.T) {
 	})
 }
 
+func TestWrapper_VerifyVP(t *testing.T) {
+	verifiableCredential := vc.VerifiableCredential{
+		Type:              []ssi.URI{ssi.MustParseURI("ExampleType")},
+	}
+	vp := vc.VerifiablePresentation{VerifiableCredential: []VerifiableCredential{verifiableCredential}}
+
+	t.Run("ok", func(t *testing.T) {
+		testContext := newMockContext(t)
+		validAt := time.Now()
+		request := VPVerificationRequest{
+			VerifiablePresentation: vp,
+			ValidAt: &validAt,
+		}
+		testContext.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
+			verifyRequest := f.(*VPVerificationRequest)
+			*verifyRequest = request
+			return nil
+		})
+		testContext.mockVerifier.EXPECT().VerifyVP(vp, true, &validAt).Return(vp.VerifiableCredential, nil)
+		testContext.echo.EXPECT().JSON(http.StatusOK, vp.VerifiableCredential)
+
+		err := testContext.client.VerifyVP(testContext.echo)
+
+		assert.NoError(t, err)
+	})
+	t.Run("ok - verifyCredentials set", func(t *testing.T) {
+		testContext := newMockContext(t)
+		verifyCredentials := false
+		request := VPVerificationRequest{VerifiablePresentation: vp, VerifyCredentials: &verifyCredentials}
+		testContext.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
+			verifyRequest := f.(*VPVerificationRequest)
+			*verifyRequest = request
+			return nil
+		})
+		testContext.mockVerifier.EXPECT().VerifyVP(vp, false, nil).Return(vp.VerifiableCredential, nil)
+		testContext.echo.EXPECT().JSON(http.StatusOK, vp.VerifiableCredential)
+
+		err := testContext.client.VerifyVP(testContext.echo)
+
+		assert.NoError(t, err)
+	})
+	t.Run("error - verification failed", func(t *testing.T) {
```

Comment:
Missing tests on other error conditions as stated in OAS

Reply:
Added tests for failed VP validation, 400 and 500 are returned by Echo